<launch> 
	<!-- Static transform between lidar_link frame and front joint frame  -->
	<node pkg= "babs_lidar_wobbler"  type= "wobbler_joint_static_tf" name= "front_wobbler_joint_static_tf" output="log" >

		<!-- Parameters for the lidar frame transform   -->
				
		<param name="wobbler_joint_name" value="front_wobbler_joint" />
		<param name="dist_to_joint_x" value="-0.925" />
		<param name="dist_to_joint_y" value="0.15" />
		<param name="dist_to_joint_z" value="0.6" />
		<param name="rot_to_joint_r" value="-0.09" />
		<param name="rot_to_joint_p" value="0.0" />
		<param name="rot_to_joint_y" value="0.0" />

		<remap from="/front_wobbler_joint_static_tf/angle" to="/front_wobbler/cmd"/>
	</node>

	<!-- Static transform between lidar_link frame and rear joint frame  -->
	<node pkg= "babs_lidar_wobbler"  type= "wobbler_joint_static_tf" name= "rear_wobbler_joint_static_tf" output="log" >

		<!-- Parameters for the lidar frame transform   -->

		<param name="wobbler_joint_name" value="rear_wobbler_joint" />
		<param name="dist_to_joint_x" value="-0.175" /> <!-- Measured with a ruler --> 
		<param name="dist_to_joint_y" value="0.15" />
		<param name="dist_to_joint_z" value="0.6" />
		<param name="rot_to_joint_r" value="-0.09" />
		<param name="rot_to_joint_p" value="0.05" />
		<param name="rot_to_joint_y" value="3.14159265359" />

		<remap from="/rear_wobbler_joint_static_tf/angle" to="/rear_wobbler/cmd"/>
	</node>

	<!-- Transform dependent on joint angle between joint and the front wobbling LIDAR lasers source.   -->
	<node pkg= "babs_lidar_wobbler"  type= "wobbler_laser_tf" name= "front_wobbler_laser_tf" output="log" > 

		<param name="wobbler_joint_name" value="front_wobbler_joint" />
		<param name="wobbler_laser_name" value="front_wobbler_laser" />

		<!-- Dummy params for node.   -->

		<param name="factor" value="1" />
		<param name="factor2" value="0.5" />

		<remap from="/front_wobbler_laser_tf/angle" to="/front_wobbler/cmd"/>
	</node>


	<!-- Transform dependent on joint angle between joint and the rear wobbling LIDAR lasers source.   -->
	<node pkg= "babs_lidar_wobbler"  type= "wobbler_laser_tf" name= "rear_wobbler_laser_tf" output="log" > 

		<param name="wobbler_joint_name" value="rear_wobbler_joint" />
		<param name="wobbler_laser_name" value="rear_wobbler_laser" />

		<!-- Dummy params for node.   -->
		
		<param name="factor" value="1" />
		<param name="factor2" value="0.5" />

		<remap from="/rear_wobbler_laser_tf/angle" to="/rear_wobbler/cmd"/>
	</node>
</launch>
